dask/bytes/core.py

```diff
@@ -141,7 +143,7 @@ def write_bytes(data, urlpath, name_function=None, compression=None,
         raise NotImplementedError("Unknown protocol for writing %s (%s)" %
                                   (protocol, urlpath))
 
-    files = open_files_write(paths, **storage_options)
+    files = [(open_files_write, p, storage_options) for p in paths]
     out = [delayed(write_block_to_file)(v, f, compression, encoding)
            for (v, f) in zip(data, files)]
```

Comment:
Ah, indeed, I forgot to include v.dask in the `Delayed(...)` object.  Should be something like the following:

```python
keys = ['write-block-%s' % tokenize(path, ...) for path in paths]
return [Delayed(key, dasks=[v.dask, {key: (write_block_to_file, v, 
                                    (apply, open_files_write, (p,), storage_options), 
                                    compression, encoding) ]) 
                for key, v, f in zip(keys, data, files)]
```